Moq.Tests/InvocationsFixture.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Moq.Tests
+{
+    public class InvocationsFixture
+    {
+	    [Fact]
+	    public void MockInvocationsAreRecorded()
+	    {
+		    var mock = new Mock<IComparable>();
+
+		    mock.Object.CompareTo(new object());
+
+		    Assert.Equal(1, mock.Invocations.Count);
+	    }
+
+		[Fact]
+	    public void MockInvocationsIncludeInvokedMethod()
+		{
+			var mock = new Mock<IComparable>();
+
+			mock.Object.CompareTo(new object());
+
+			var invocation = mock.Invocations[0];
+
+			var expectedMethod = typeof(IComparable).GetMethod(nameof(mock.Object.CompareTo));
+
+			Assert.Equal(expectedMethod, invocation.Method);
+		}
+
+	    [Fact]
+	    public void MockInvocationsIncludeArguments()
+	    {
+			var mock = new Mock<IComparable>();
+
+		    var obj = new object();
+
+			mock.Object.CompareTo(obj);
+
+		    var invocation = mock.Invocations[0];
+
+		    var expectedArguments = new[] {obj};
+
+			Assert.Equal(expectedArguments, invocation.Arguments);
+	    }
+
+	    [Fact]
+	    public void MockInvocationsCanBeEnumerated()
+	    {
+			var mock = new Mock<IComparable>();
+
+		    mock.Object.CompareTo(-1);
+		    mock.Object.CompareTo(0);
+		    mock.Object.CompareTo(1);
+
+		    var count = mock.Invocations.Count();
```

Comment:
I understand what you're doing here, but I wonder if this could be implemented more clearly. What if LINQ's `.Count()` operator has special handling for `IReadOnlyList<>` (as requested by e.g. https://github.com/dotnet/corefx/issues/24773)? Then this possibly wouldn't enumerate at all! If the test is about enumeration, it would be good to have the enumeration more visible.

Reply:
Fair point. I'll break it out into a `foreach` to be certain.

Reply:
Actually to be 100% certain, I'll call the enumerator manually. I know `foreach` has some special behaviours for lists.